### ISSUE TITLE
Focus sometimes switches to the search box in the help pane after executing `?` in the console

### ISSUE THREAD

**Issue by ha0ye** [rstudio/rstudio#12741]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2022.12.0+353
    OS Version      : macOS Monterey 12.6.2
    R Version       : 4.1.1

### Steps to reproduce the problem

Sometimes when executing `?length` or any function name in the console, the help page shows up in the panel, and then focus switches to the search bar of the help panel.

It is not clear under what conditions this happens.

@prubin73 has experienced the same issue, but is similarly unable to 100% reproduce the conditions under which this happens. https://mathstodon.xyz/@parubin/109910133733518479

### Describe the problem in detail

No further details, but it seems like this may be similar to #10600 ?

### Describe the behavior you expected

It would be ideal if focus did not switch unless through action of the user.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by prubin73** [comment]
I think I may have identified a pattern. Initially, regardless of whether I have files open or not and whether or not I have executed any other commands (either from a file or from the console), typing "?whatever" in the console leaves the cursor in the console. Once I interact with the help window (scroll an entry either direction, use the back or forward arrow buttons, but _not_ click the home button) and return the cursor to the console, subsequent help requests in the console shift the cursor to the help window search box. This is true for both "?" and "help(...)".

**Comment by ha0ye** [comment]
Thanks @prubin73 - I can confirm this as a set of minimal reproducible conditions.

If I type text in the search bar in the help, but do NOT hit enter to engage the search, it is not sufficient as a trigger to draw focus on later instances of "?" in the console.

What a bizarre bug!

**Comment by ronblum** [comment]
@ha0ye Thank you for raising this and brining it over from Mastodon, and thank you for providing the minimal example! We'll review this as we continue development of RStudio.

I can also reproduce this in
- RStudio Desktop 2022.07.2+576 and 2023.03.0-daily+351 on MacOS 13.2
- RStudio Server 2023.03.0-daily+347 on Ubuntu 20.04 via Chrome on MacOS 13.2

**Comment by gtritchie** [comment]
Research note: Likely introduced by this: https://github.com/rstudio/rstudio/pull/7408

https://github.com/rstudio/rstudio/blob/df42e153889de55c6025ed832105b3dd9bd5a2e6/src/gwt/src/org/rstudio/studio/client/workbench/views/help/HelpPane.java#L230-L233

**Comment by ronblum** [comment]
Verified in
- RStudio Desktop 2023.05.0-daily+92 on MacOS 13.2
- RStudio Server 2023.05.0-daily+92 on Ubuntu 20.04.2 via Chrome on MacOS 13.2
